datafusion/core/src/physical_optimizer/join_pipeline_selection.rs

```diff
@@ -435,6 +509,123 @@ pub fn swap_sliding_nested_loop_join(
+    // Extract relevant properties from the hash join plan
+    let filter = hash_join.filter();
+    let left_order = hash_join.left().output_ordering();
+    let right_order = hash_join.right().output_ordering();
+    let is_left_unbounded = is_plan_streaming(hash_join.left());
+    let is_right_unbounded = is_plan_streaming(hash_join.right());
+
+    // Based on various properties of the join and the data streams, determine the
+    // best way to process the hash join.
+    match (
+        is_left_unbounded,
+        is_right_unbounded,
+        filter,
+        left_order,
+        right_order,
+    ) {
+        // Both streams are unbounded, and filter with orders are present
+        (true, true, Some(filter), Some(left_order), Some(right_order)) => {
+            // Check if filter expressions can be pruned based on the data orders
+            let (build_prunable, probe_prunable) = is_filter_expr_prunable(
+                filter,
+                Some(left_order[0].clone()),
+                Some(right_order[0].clone()),
+                || hash_join.left().equivalence_properties(),
+                || hash_join.left().ordering_equivalence_properties(),
+                || hash_join.right().equivalence_properties(),
+                || hash_join.right().ordering_equivalence_properties(),
+            )?;
+
+            // Extract details from the aggregate parent plan
+            let AggregateExec {
+                mode,
+                group_by,
+                aggr_expr,
+                ..
+            } = parent_plan;
+
+            // TODO: Implement FIRST_VALUE convert into LAST_VALUE.
+            let fetch_per_key =
+                if aggr_expr.iter().all(|expr| expr.as_any().is::<LastValue>()) {
+                    1
+                } else {
+                    return Ok(None);
+                };
+
+            // If probe side can be pruned, apply specific optimization for that case
+            if probe_prunable
+                && matches!(mode, AggregateMode::Partial)
+                && matches!(hash_join.join_type(), JoinType::Inner | JoinType::Right)
+                && group_by.null_expr().is_empty()
+            {
+                // Create a new sliding hash join plan
+                let sliding_hash_join = Arc::new(PartitionedHashJoinExec::try_new(
+                    hash_join.left.clone(),
+                    hash_join.right.clone(),
+                    hash_join.on.clone(),
+                    filter.clone(),
+                    &hash_join.join_type,
+                    hash_join.null_equals_null,
+                    left_order.to_vec(),
+                    right_order.to_vec(),
+                    fetch_per_key,
+                )?);
+                return Ok(Some(vec![sliding_hash_join]));
+            // If build side can be pruned, apply specific optimization for that case
+            } else if build_prunable
+                && matches!(mode, AggregateMode::Partial)
```

Comment:
```suggestion
                &&  mode.is_first_stage()
```
although `is_first_stage` is not public. I will fix it